seabug_contracts/CallContract.purs

```diff
@@ -0,0 +1,276 @@
+  ( ContractConfig(ContractConfig)
+  , defaultSlotConfig
+  , mkDatumCacheWebSocketAff
+  , mkOgmiosWebSocketAff
+  , runContract
+  , runContract_
+  )
+import Contract.Numeric.Natural (toBigInt)
+import Contract.Prelude
+  ( Aff
+  , Either
+  , Maybe(Just)
+  , fromMaybe
+  , liftEither
+  , liftM
+  , note
+  , pure
+  , unwrap
+  , wrap
+  , (<$>)
+  , (<<<)
+  , (<>)
+  , (=<<)
+  , (>>>)
+  , show
+  )
+import Contract.Prim.ByteArray
+  ( byteArrayFromString
+  , byteArrayToHex
+  , hexToByteArray
+  )
+import Contract.Scripts
+  ( ed25519KeyHashFromBech32
+  , ed25519KeyHashToBech32Unsafe
+  , scriptHashFromBech32
+  , scriptHashToBech32Unsafe
+  )
+import Contract.Transaction
+  ( TransactionInput(TransactionInput)
+  , TransactionOutput(TransactionOutput)
+  )
+import Contract.Value
+  ( Value
+  , getCurrencySymbol
+  , getNonAdaAsset
+  , getTokenName
+  , mkCurrencySymbol
+  , mkTokenName
+  , valueToCoin
+  )
+import Control.Bind (bind)
+import Control.Promise (Promise)
+import Control.Promise as Promise
+import Data.Array (singleton)
+import Data.BigInt (BigInt)
+import Data.BigInt as BigInt
+import Data.Function (($), (#))
+import Data.List (toUnfoldable)
+import Data.Tuple.Nested ((/\))
+import Data.UInt as UInt
+import Data.Unit (Unit)
+import Effect (Effect)
+import Effect.Aff (error)
+import Effect.Class (liftEffect)
+import Effect.Exception (Error)
+import Metadata.Seabug (SeabugMetadata(SeabugMetadata))
+import Metadata.Seabug.Share (unShare)
+import Seabug.Contract.MarketPlaceBuy (marketplaceBuy)
+import Seabug.Contract.MarketPlaceListNft (ListNftResult, marketPlaceListNft)
+import Seabug.Types (NftCollection(NftCollection), NftData(NftData), NftId(NftId))
+import Serialization.Address (addressBech32, intToNetworkId)
+import Types.Natural as Nat
+import Types.Value (flattenNonAdaValue)
+import UsedTxOuts (newUsedTxOuts)
+import Wallet (mkNamiWalletAff)
+
+-- | Exists temporarily for testing purposes
+callMarketPlaceBuyTest :: String -> Effect (Promise String)
+callMarketPlaceBuyTest str = Promise.fromAff do
+  pure str
+
+-- | Calls Seabugs marketplaceBuy and takes care of converting data types.
+--   Returns a JS promise holding no data.
+callMarketPlaceBuy :: ContractConfiguration -> BuyNftArgs -> Effect (Promise Unit)
+callMarketPlaceBuy cfg args = Promise.fromAff do
+  contractConfig <- buildContractConfig cfg
+  nftData <- liftEffect $ liftEither $ buildNftData args
+  runContract_ contractConfig (marketplaceBuy nftData)
+
+-- | Calls Seabugs marketPlaceListNft and takes care of converting data types.
+--   Returns a JS promise holding nft listings.
+callMarketPlaceListNft :: ContractConfiguration -> Effect (Promise (Array ListNftResultOut))
+callMarketPlaceListNft cfg = Promise.fromAff do
+  contractConfig <- buildContractConfig cfg
+  listnft <- runContract contractConfig marketPlaceListNft
+  pure $ buildNftList <$> listnft
+
+-- | Configuation needed to call contracts from JS.
+type ContractConfiguration =
```

Comment:
I suppose this will work, but why not just take `ServerConfig`s, which we've already defined? It would make it easier to update the `defaultContractConfig` as well

Reply:
I can do it, but also this way the user-facing structure of the config is independent of how we want to keep things in Purescript modules. Should I go with ServerConfig instead?